src/testing/tree_construction/parser.rs

```diff
@@ -262,7 +262,7 @@ fn document(i: Span) -> IResult<Span, Span> {
 }
 
 fn document_fragment(i: Span) -> IResult<Span, Span> {
-    preceded(tag("#document-fragment\n"), take_until1("#"))(i)
+    preceded(tag("#document-fragment\n"), take_until1("\n"))(i)
```

Comment:
It seems that we have to capture until the newline instead of the next line's `#`.

I thought this would mess up tests, but it turns out this works fine.